eng/common/scripts/Package-Properties.ps1

```diff
@@ -164,21 +159,45 @@ function Get-CSVMetadata ([string]$MetadataUri=$MetadataUri)
     return $metadataResponse
 }
 
-function Operate-OnPackages ($activePkgList, $ServiceDirectory, [Array]$pkgPropsResult)
+function Get-PkgPropsForEntireService ($serviceDirectoryPath, $serviceDirectoryName)
 {
-    foreach ($pkg in $activePkgList)
+    $projectProps = @() # Properties from very project inthe service
+    $packageProps = @() # Properties for artifacts specified in ci.yml
+
+    foreach ($directory in (Get-ChildItem $serviceDirectoryPath -Directory))
+    {
+        $pkgDirectoryPath = Join-Path $serviceDirectoryPath $directory.Name
+
+        if ($GetPackageInfoFromRepoFn -and (Test-Path "Function:$GetPackageInfoFromRepoFn"))
+        {
+            $pkgProps = &$GetPackageInfoFromRepoFn -pkgDirectoryPath $pkgDirectoryPath -serviceDirectory $serviceDirectoryName
+            if ($null -ne  $pkgProps)
+            {
+                $projectProps += $pkgProps
+            }
+        }
+        else
+        {
+            LogError "The function for '$GetPackageInfoFromRepoFn' was not found.`
+            Make sure it is present in eng/scripts/Language-Settings.ps1 and referenced in eng/common/scripts/common.ps1.`
+            See https://github.com/Azure/azure-sdk-tools/blob/master/doc/common/common_engsys.md#code-structure"
+        }
+    }
+
+    $ciYmlFiles = Get-ChildItem $serviceDirectoryPath -filter "ci.*yml"
```

Comment:
Now that we are parsing them from the directory do we neve need to parse the ci.yml files at all?

Reply:
If we do care about the ci.yml filtering still we should loop over the ciYmlFiles first and then use that artifact name and continue passing it into the GetPackageInfoFromRepoFn like we were before.

Reply:
Well that's what we were doing before. I thought you wanted us to get all the packages first then filter on the artifact list. At least that's what I understood from you previous comment except maybe am reading it wrong.

> I think it might make more sense to loop over the directory and open all the project files once and then build a list based on that and then find the ones that are applicable to artifact list.

> So perhaps we have a function that essentially parses the project for all the directories in a service directory and then we post process those to try and align the artifact names to those packages.